pkg/utils/subprocesses.go

```diff
@@ -0,0 +1,36 @@
+package utils
+
+import "sync"
+
+// Subprocesses is an abstraction over the following pattern:
+//
+//   var wg sync.Subprocesses
+//   wg.Add(1)
+//   go func() {
+//     defer wg.Done()
+//     // ...
+//   }()
+//
+// Which becomes:
+//
+//  var subs utils.Subprocesses
+//  subs.Go(func() {
+//     // ...
+//  })
+type Subprocesses struct {
+	wg sync.WaitGroup
+}
+
+// Wait blocks until all function calls from the Go method have returned.
+func (s *Subprocesses) Wait() {
+	s.wg.Wait()
+}
+
+// Go calls the given function in a new goroutine.
+func (s *Subprocesses) Go(f func()) {
+	s.wg.Add(1)
+	go func() {
+		defer s.wg.Done()
+		f()
+	}()
+}
```

Comment:
I disagree. The main idea of `Subprocesses` is that users CAN'T `Add()` more or less than they `Done()`. 
This happens to me a lot and it's a pain to debug. With your solution, users can still get into this situation.

Reply:
In that case, IMHO the docs should say more about how to use this properly without assuming any knowledge of WaitGroup.  For example, a naive user that calls Go() after Wait() would cause a race. Alternatively, if this was moved to package monitoring, then I think it's fine as-is - I'm just overly concerned about keeping _pkg/utils_ API super clean/stable.

Reply:
That's fair. I'll move it to pkg/monitoring